testplan/runners/pools/remote.py

```diff
@@ -535,23 +544,26 @@ def _write_syspath(self):
 
     def starting(self):
         """Start a child remote worker."""
-        if self.parent.pool:
-            self.cfg.async_start = True
-            self.parent.pool.apply_async(self._starting_impl())
-        else:
-            self._starting_impl()
-
-    def _starting_impl(self):
         self._prepare_remote()
         super(RemoteWorker, self).starting()
 
     def stopping(self):
         """Stop child process worker."""
         self._fetch_results()
-        if self.cfg.pull:
-            self._pull_files()
         super(RemoteWorker, self).stopping()
 
+    def _wait_stopped(self, timeout=None):
+        sleeper = get_sleeper(1, 300)
+        while next(sleeper):
+            if self.status.tag != self.status.STOPPED:
+                self.logger.info('Waiting for workers to stop')
+            else:
+                break
+        else:
+            self.logger.warning(
```

Comment:
Maybe we should raise an exception at this point instead of just continuiing?

Reply:
I think by the time we are stopping, we should best effort stop everything that we can. If we raise here or abort, we are not cleaning up as much as possible. But this is indeed not very neat...

The problem is starting and stopping could overlap. Some treadmill worker could come up and starting executing tests and even finished all tests before other workers are still initializing. I'm thinking whether if we should be more rigid to have all resources readily started before we start to execute test.

Reply:
hmmm, i'm rethinking this. I'm going to make some change to treadmill pool so that start/stop does not overlap.

Reply:
I usually prefer to fail noisily (raise an Exception/abort etc.) if something unexpected happens, as that can make it easier to debug - rather than leaving your application in a weird inconsistent state that causes an issue far away and having to search back to find a relevant log!